RELEASE.md

```diff
@@ -75,24 +75,26 @@ you can skip the dependency update or only update select dependencies. In such a
 case, you have to create an issue or pull request in the GitHub project for
 later follow-up.
 
-### Prepare your release
+### 1. Prepare your release
 
-For a patch release, work in the branch of the minor release you want to patch.
+For a new major or minor release, create the corresponding release branch based on the master branch. Note that a new a new release branch in `origin` is only necessary if there is not already one for this <major>.<minor> version. For example if we're releasing 2.17.0 and the previous stable release is 2.16.0 we need to create a `release-2.17` branch. Patch release don't require their own `release-<major>.<minor>.<patch>, and the same for release candidate versions.
 
-For a new major or minor release, create the corresponding release branch based on the master branch.
+For a patch release or release candidate any changes should be merged into the previously mentioned release branch. So for `2.16.1` or `2.16.0-rc.1` create a new branch off of `release-2.16` and open a PR with the necessary bug/regression fix.
```

Comment:
To be honest, mentioning new branch creation here confuses me more than it enlightens me. "open a PR" already implies that you have a personal branch, either in the repo or, if you prefer, in a clone of the repo. Also, that branch and PR is not necessarily created by the release shepherd, but by whoever has worked on the bug fix, prior to cutting the RC or patch release. I would probably get rid of this paragraph, as it suggests patch releases or RCs are special in this regard, which is not really the case.